openglyph/include/openglyph/io/mega_file.hpp

```diff
@@ -0,0 +1,74 @@
+#pragma once
+
+#include <khepri/io/exceptions.hpp>
+#include <khepri/io/file.hpp>
+#include <khepri/io/stream.hpp>
+
+#include <cstdint>
+#include <string>
+#include <tuple>
+#include <vector>
+
+namespace openglyph::io {
+
+/**
+ * @brief Represents a single MegaFile archive.
+ *
+ * This class provides functionality to open files stored inside
+ * a MegaFile archive (.meg). It manages the archive's metadata,
+ * including filenames and file info necessary for accessing
+ * individual files within the archive.
+ *
+ * Instances of this class are non-copyable and non-movable.
+ */
+class MegaFile final
+{
+public:
+    explicit MegaFile(const std::filesystem::path& mega_file_path);
+    ~MegaFile() = default;
+
+    MegaFile(const MegaFile&)                = delete;
+    MegaFile& operator=(const MegaFile&)     = delete;
+    MegaFile(MegaFile&&) noexcept            = delete;
+    MegaFile& operator=(MegaFile&&) noexcept = delete;
+
+    /**
+     * @brief Opens a file from the MegaFile archive.
+     *
+     * This method attempts to locate and open a file within the archive
+     * using the provided relative path. It first calculates the CRC32 hash
+     * of the path and uses it to look up the file's metadata in the archive.
```

Comment:
How it opens a file doesn't matter. Remove the stuff about CRC calculation here. Just describe the behavior relevant for the caller.